mlir/lib/Dialect/SPIRV/IR/SPIRVTypes.cpp

```diff
@@ -184,9 +181,7 @@ void CompositeType::getCapabilities(
             .getCapabilities(capabilities, storage);
       })
       .Case<TensorArmType>([&](TensorArmType type) {
-        static const Capability caps[] = {Capability::TensorsARM};
-        ArrayRef<Capability> ref(caps, std::size(caps));
-        capabilities.push_back(ref);
+        capabilities.push_back({Capability::TensorsARM});
```

Comment:
IIUC, this creates dangling references. `ArrayRef` stores pointers, so you need to push back something with a static storage duration here (`static constexpr` variable).